### ISSUE TITLE
Inconsistent Segmentation Results When Using device=mps Compared to cpu and cuda

### ISSUE THREAD

**Issue by daisy9542** [ultralytics/ultralytics#16827]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

I have fine-tuned the same YOLO model and noticed inconsistent behavior when using different devices. Specifically, when using device=cpu or device=cuda, the prediction results are consistent. However, when using device=mps, the segmentation  appears to remove the bounding boxes, showing only the incorrect segmentation masks.

Below is my command:

```bash
yolo predict model=[model_path] source=[source_path] imgsz=1280 retina_masks=True show_boxes=False device=[device]
```

- device=cuda

![image_result_cuda](https://github.com/user-attachments/assets/2ec221e5-8af9-441d-bfd8-54194abdba5e)

```txt
0 0.447656 0.254167 0.447656 0.263889 0.448438 0.263889 0.449219 0.265278 0.45 0.265278 0.450781 0.266667 ...
```

- device=cpu

![image_result_cpu](https://github.com/user-attachments/assets/109cdbf7-95c3-4583-9c40-369a9dd59e59)

```txt
0 0.447656 0.254167 0.447656 0.263889 0.448438 0.263889 0.449219 0.265278 0.45 0.265278 0.450781 0.266667 ...
```

- device=mps

![image_result_mps](https://github.com/user-attachments/assets/ac56d2a3-bb40-4c45-8a6b-14714dc76aed)

```txt
0 0.449219 0.265278 0.45 0.265278 0.450781 0.266667 0.451562 0.266667 0.453125 0.269444 0.454688 0.269444 ...
```

## Environment:

- **Operating System**: NVIDIA GeForce RTX 4090 (cuda), Apple M1 macOS Sonoma 14.6.1 (cpu and mps)
- **YOLO Version**: YOLO11 (finetune on cuda)
- **Venv Version**: python 3.9, ultralytics  8.2.103, torch 2.3.1, torchvision 0.18.1, numpy 1.23.2

Could you please tell me why the mps device behaves differently, specifically removing bounding boxes in the segmentation output? Any guidance or fixes would be greatly appreciated.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @daisy9542, thank you for reaching out and using Ultralytics 🚀! This is an automated response, but don't worry—an Ultralytics engineer will assist you soon.

We recommend visiting our comprehensive [Docs](https://docs.ultralytics.com) which include various [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples that might provide immediate solutions.

Since this relates to device-specific behavior, if it's a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) to help us debug the issue more effectively.

For further custom training ❓ Questions, please ensure to include detailed information like dataset examples or training logs and make sure you're following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join our community and choose your favorite platform! For instant chats, hop on [Discord](https://ultralytics.com/discord) 🎧. For in-depth discussions, visit [Discourse](https://community.ultralytics.com), and if you enjoy Reddit, join us on our [Subreddit](https://reddit.com/r/ultralytics).

## Upgrade

Please upgrade to the latest version of the `ultralytics` package with all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment using [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to ensure the issue isn't already resolved:

```bash
pip install -U ultralytics
```

## Environments

YOLO runs smoothly in these verified environments, all dependencies (like [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/), [PyTorch](https://pytorch.org/)) included:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) across platforms every 24 hours and upon every commit.

Thank you for your patience and contribution! 😊

**Comment by Y-T-G** [comment]
It could be a bug in PyTorch.
